### ISSUE TITLE
batch Video Upscaling script

### ISSUE THREAD

**Issue by zelenooki87** [ArcticHare105/S3Diff#11]
I've written a script that performs video upscaling using S3Diff in batch mode. The file inference_s3diff_video.py should be located in the src subdirectory of the S3Diff folder. The default location for input video files is the "video" folder. Place the video upscaling script, video_upscale.py, in the root directory of S3Diff. The models are assumed to be at these paths: 'assets/mm-realsr/de_net.pth', 'models/s3diff.pkl', and 'models/sd-turbo'. You can modify the paths to the models within the upscaling script itself, as well as the codec if you prefer a different one. :)

Simply run the script within your conda environment with the command: python video_upscale.py

The upscaled videos will be placed in the "video" folder alongside the originals, with a suffix appended. :)

You'll need to download and install FFmpeg. On Windows, you'll need to add it to your PATH environment variable. I haven't tested the script on Ubuntu, but it should work. :)
[S3Diff.zip](https://github.com/user-attachments/files/17444631/S3Diff.zip)

**Comment by ArcticHare105** [comment]
@zelenooki87 Thank you so much for your valuable contribution!

**Comment by zelenooki87** [comment]
:) I know the model isn't intended for video upscaling and doesn't give perfect results, but it gives okay and more consistent results, and better than, for example, OseDiff. I just forgot to mention that the sr_test.yaml path needs to be adjusted: lr_path: input. I'm glad if it helps someone :)

**Comment by zelenooki87** [comment]
I've added the edited script and the necessary files to the first message. :)

**Comment by MarcoRavich** [comment]
@Selur may help too, but video upscaling needs more "consistencies" than image: https://videogigagan.github.io/

**Comment by Selur** [comment]
Yes, unless you reencode animes/cartoons or similar, temporal consistency is a problem when using image upscalers.
Sadly, afaik, there is no project which uses VideoGigaGan and provides any pretrained models.
In general, for image models, I would suggest to use [ChainNer ](https://github.com/chaiNNer-org/chaiNNer) or through Vapoursynth if the model can be converted to .onnx  (for https://github.com/AmusementClub/vs-mlrt) or .pth (for https://github.com/rlaphoenix/VSGAN). (personally, I prefer using Vapoursynth since it offers tons of additional stuff and can be easily fed to most encoders)
